store/iavl/store.go

```diff
@@ -43,7 +43,14 @@ type Store struct {
 // store's version (id) from the provided DB. An error is returned if the version
 // fails to load.
 func LoadStore(db dbm.DB, id types.CommitID, lazyLoading bool) (types.CommitKVStore, error) {
-	tree, err := iavl.NewMutableTree(db, defaultIAVLCacheSize)
+	return LoadStoreWithInitialVersion(db, id, lazyLoading, 0)
+}
+
+// LoadStore returns an IAVL Store as a CommitKVStore setting its initialVersion
+// to the one given. Internally, it will load the store's version (id) from the
+// provided DB. An error is returned if the version fails to load.
+func LoadStoreWithInitialVersion(db dbm.DB, id types.CommitID, lazyLoading bool, initialVersion uint64) (types.CommitKVStore, error) {
```

Comment:
Note that this will probably have to be modified once IAVL 0.15 is released. In 0.14, loading a missing version from an _empty_ tree (no versions) does not error, which has been changed/fixed in 0.15: https://github.com/cosmos/iavl/pull/317

Reply:
Thanks for the update. We'll just have to update this once that version is released.